### ISSUE TITLE
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './v4' is not defined by "exports" package.json

### ISSUE THREAD

**Issue by CSharpDaddy** [uuidjs/uuid#594]
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './v4' is not defined by "exports" in C:\Users\jaden\Desktop\Hye\node_modules\uuid\package.json

Does anyone know why this is happening? Im using the latest node.js version

**Comment by LinusU** [comment]
Are you doing `import ... from 'uuid/v4'` anywhere in your code?

**Comment by ctavan** [comment]
Please check the discussion in #584. Issues like this are with absolute certainty due to badly defined dependency version ranges. Please make sure that your code and all dependencies you are using make proper use of Semantic Versioning version ranges.

The whole purpose of [Semantic Versioning](https://semver.org/) is to provide library authors with a way for safely introducing breaking changes, and developers with a safe way of dealing with such changes by specifying appropriate dependency version ranges.

If we start supporting legacy forever just because some other library or application specifies dependencies in a wrong way it would defeat the original purpose of Semantic Versioning.

We will therefore not start fixing other people's code by re-introducing technical debt.

**Comment by tchakabam** [comment]
> "and all dependencies you are using " 

even if one finds that they aren't doing this right in any deps, do you expect people to fork all these deps etc? :)

this is what i meant by idealistic concepts in another post on SO regarding this :)

yes, you can break things in major versions. in theory :) because you also expect here that people will use ranges in the according way. which in practical reality (unfortunately) is not the case.

> We will therefore not start fixing other people's code by re-introducing technical debt.

i dont know what you mean by "technical debt" here. to me all this is just syntactical sugar for one or the other way to express a JS import honestly :D it's not really a feature of any library itfp i would say.

i get your point philosophicaly but i dont find it very pragmatic.

either way, i dont care what is done here. i already said that i found a solution to the problem. i just wanted to help you with providing the data i posted before.

you dont need to come up with grand justifications for how you proceed on the matter.

**Comment by LinusU** [comment]
> even if one finds that they aren't doing this right in any deps, do you expect people to fork all these deps etc? :)

Realistically, it's probably only one package that accidentally used the wrong range. Should be as easy to fix that package with a PR, as sending a PR to this repo.

> [...] because you also expect here that people will use ranges in the according way. which in practical reality (unfortunately) is not the case.

I think that this is a very very rare occurrence. Npm automatically uses the correct ranges when you run `npm install`, and I've never worked in a project that doesn't use ranges.

Are you really suggeting that no package on npm can ever make any breaking changes?

> either way, i dont care what is done here. i already said that i found a solution to the problem for our case. i just wanted to help you with providing the data i posted before.

Something that would really help is seeing which package is actually broken. Could you run `npm ls uuid` and post the result here?

**Comment by broofa** [comment]
@tchakabam: I always appreciate it when someone submits a PR for us.  That's more than 95% of people do,  so thank you for that.  However, as the self-appointed curmudgeon on this project, I'm compelled to make a few observations about how you're coming across here...

### Failure to provide reproduceable test case

@ctavan [asked for this](https://stackoverflow.com/questions/62549457/error-err-package-path-not-exported-package-subpath-v4-is-not-defined-by/69564152?noredirect=1#comment123958765_69564152), which you haven't done.  In terms of understanding the problem you're complaining about, all you've done is provide some excerpts of your package-lock.json file, which has not been helpful.

There's a reason [we are very explicit about needing reproducible test cases](https://github.com/uuidjs/uuid/blob/main/.github/ISSUE_TEMPLATE/bug_report.md?plain=1#L18-L20):  It avoids unnecessary confusion and drama.  Oftentimes, it reveals the issue isn't what people thought.  Or it leads to a better understanding of the problem that reveals why the code is the way it is.  Or, best of all, it provides real insight into what the Right Solution™ should be and we can work with you to get a fix in.

Ignoring such requests says you don't actually care about fixing the problem. 

### Snarky comments

>  [you're welcome](https://github.com/uuidjs/uuid/pull/584#issuecomment-985710103)

This sort of passive-aggressive nonsense will never be well received.

### Willful ignorance

> [i am not claiming to know what the issue is, or that it is necessarily to be fixed on uuid project side.](https://github.com/uuidjs/uuid/pull/584#issuecomment-984102344)

If you're not sure, do the work needed to figure it out.  Or at least be humble about recognizing that your understanding is incomplete and might be wrong.  Aside from this one comment, you seem bent on loudly and repeatedly claiming to be right, despite our attempts to help you.

### General dismissiveness

I'm not gonna beat around the bush on this one.  Your dismissal of the decisions and rationale for how we do things here as [idealistic concepts](https://github.com/uuidjs/uuid/issues/594#issuecomment-985708609rl) is just laughable and intellectually lazy.  That Christoph and Linus continue to be polite about that fact is remarkable.

TL;DR: Congrats on motivating me to google how to [block users from github repos](https://github.blog/2016-04-04-organizations-can-now-block-abusive-users/).  Please, let's not have it get to that point.
